### DISCUSSION TITLE
JSON.parse() Unhandled Runtime Error

### DISCUSSION THREAD

**Discussion by Chandler-Zhu** [vercel/next.js#40027] [Help]
repro link: https://codesandbox.io/s/peaceful-sun-htc9vg?file=/pages/index.js&resolutionWidth=787&resolutionHeight=675
The problem I am having is when I go to a pokemon detail page and navigate back, the name of the pokemon does not match the image anymore
before: 
![image](https://user-images.githubusercontent.com/61914365/187076926-45210a7e-0619-4241-987d-cdda2cd62245.png)

after:
![image](https://user-images.githubusercontent.com/61914365/187076940-f55f56f9-d590-41f5-977f-4ef0b0e4b0f4.png)

also when I click the 'prev' button it gives me an error
![image](https://user-images.githubusercontent.com/61914365/187076985-8562e1b6-31c7-4e2e-97ee-93e1e03a3d28.png)

**Comment by icyJoseph** [comment]
Did you fix this issue? I cannot reproduce it on the link you gave.

**Reply by Chandler-Zhu** [reply]
1.click 'next' a few times make sure you're not in the first page
2.click any pokemon to go to the detail page
3.navigate back using back button or clicking the header
![image](https://user-images.githubusercontent.com/61914365/187248995-571314e3-cbec-4c24-99e4-afe4c2c6a065.png)

**Reply by icyJoseph** [reply]
Aha, I see, the `initialPokemon` props, are what's driving the issue, so to speak. When you return to the landing page, your valtio state is at the offset, but the react state 'starts again' from `initialPokemon`.

**Reply by Chandler-Zhu** [reply]
should I just use localStorage instead ?
